save-points/05-add-authentication/BlazingPizza.Client/Pages/Index.razor

```diff
@@ -39,47 +38,37 @@
         <div class="empty-cart">Choose a pizza<br>to get started</div>
     }
 
-    <div class="order-total @(OrderState.Order.Pizzas.Any() ? "" : "hidden")">
+    <div class="order-total @(Order.Pizzas.Any() ? "" : "hidden")">
         Total:
-        <span class="total-price">@OrderState.Order.GetFormattedTotalPrice()</span>
-        <button class="btn btn-warning" disabled="@(OrderState.Order.Pizzas.Count == 0)" onclick="@PlaceOrder">
+        <span class="total-price">@Order.GetFormattedTotalPrice()</span>
+        <button class="btn btn-warning" disabled="@(Order.Pizzas.Count == 0)" onclick="@PlaceOrder">
             Order >
         </button>
     </div>
 </div>
 
 @if (OrderState.ShowingConfigureDialog)
 {
-    <ConfigurePizzaDialog />
+    <ConfigurePizzaDialog
+        Pizza="@OrderState.ConfiguringPizza"
+        OnConfirm="@OrderState.ConfirmConfigurePizzaDialog" 
+        OnCancel="@OrderState.CancelConfigurePizzaDialog" />
 }
 
 @functions {
-    [CascadingParameter] UserStateProvider UserState { get; set; }
-
     List<PizzaSpecial> specials;
+    Order Order => OrderState.Order;
 
     protected async override Task OnInitAsync()
     {
-        OrderState.StateChanged += OnOrderStateChanged;
         specials = await HttpClient.GetJsonAsync<List<PizzaSpecial>>("specials");
     }
 
-    void IDisposable.Dispose()
-    {
-        OrderState.StateChanged -= OnOrderStateChanged;
-    }
-
-    void OnOrderStateChanged(object sender, EventArgs e) => StateHasChanged();
-
     async Task PlaceOrder()
     {
-        // The server will reject the submission if you're not signed in, so attempt
-        // to sign in first if needed
-        if (await UserState.TrySignInAsync())
```

Comment:
It's difficult to interpret all the changes in the flow, but could you clarify why we're removing the `TrySignInAsync` from here and in other places?

Reply:
Good catch. This is indeed a mistake.